### ISSUE TITLE
Plan for the maintenance of org.eclipse.egit.github.core

### ISSUE THREAD

**Issue by merks** [eclipse-mylyn/org.eclipse.mylyn#133]
The source of this plugins is here: 

https://git.eclipse.org/c/egit/egit-github.git/tree/org.eclipse.egit.github.core

But all the other projects have migrated to Mylyn, except this one project.

Do we really expect the EGit project to keep maintaining this one essential bundle?  Their update site doesn't even include a source bundle so it's really hard to work with this thing and it contains lots of really useful functionality.

https://download.eclipse.org/egit/github/updates/

 It's already included in Mylyn's own update site, so wouldn't it be better if this bundle were maintained by the Mylyn project?

**Comment by ruspl-afed** [comment]
@BeckerFrank  @wimjongman , I agree with Ed @merks.  The mentioned bundle has no dependencies to the rest of EGit and defines API for GiHub. After migrating others, it seems natural to move this to Mylyn as well.

**Comment by BeckerFrank** [comment]
We got the following mail:
Von: Wim Jongman <notifications@github.com>
Betreff: Aw: [eclipse-mylyn/org.eclipse.mylyn] "Appropriate" GitHub connector in mylyn.tasks (Issue #18)
Datum: 20. Januar 2023 um 12:16:13 MEZ
An: "eclipse-mylyn/org.eclipse.mylyn" <org.eclipse.mylyn@noreply.github.com>
Kopie: Frank Becker <eclipse@frank-becker.de>,  Author <author@noreply.github.com>
Antwort an: "eclipse-mylyn/org.eclipse.mylyn" <reply+ADEPFUH7PKFEIFANX6KXREWB26VH3EVBNHHFUSNZLI@reply.github.com>


Thomas Wolf said on the Mylyn-dev list:
Sat, 7 Jan, 09:35 (13 days ago)
to Matthias, Mylyn, EGit On 06.01.23 22:19 , Matthias Sohn wrote:
We didn't yet discuss in the EGit project if we want to take up maintenance again since Mylyn is revived or if we hand it over to the Mylyn project.
[@eGit](https://github.com/eGit) committers: what do you think?
Sure, if Mylyn wants to maintain and improve it, that's perfectly fine with me. We never did much with it. I think the only new development in the past couple of years in the Mylyn connector was my changing the repository set-up dialog to support GitHub PATs better.
However, see below:
	•	Keep org.eclipse.egit.github.core where it currently is; don't move that one to Mylyn.
	•	Also move org.eclipse.egit.mylyn.ui to Mylyn.
On Fri, Jan 6, 2023 at 3:46 PM Wim Jongman <[wim.jongman@gmail.com](mailto:wim.jongman@gmail.com) [mailto:[wim.jongman@gmail.com](mailto:wim.jongman@gmail.com)](mailto:%5Bwim.jongman@gmail.com%5D(mailto:wim.jongman@gmail.com))> wrote:
Matthias, we are moving the EGit GitHub connector into Mylyn. WDYT?

PR is here:
https://github.com/eclipse-mylyn/org.eclipse.mylyn/pull/38
<https://github.com/eclipse-mylyn/org.eclipse.mylyn/pull/38>
This PR also copies org.eclipse.egit.github.core, which is not part of the Mylyn connector. It is a completely separate thing and does not depend on Mylyn or EGit at all.
I would prefer that one to stay where it is, in the org.eclipse.egit.github repository. Reasons:
	1.	It is independent.
	2.	It uses a BREE of JAVA 8 for Android compatibility.
	3.	It should be versioned, built, tested, and published independently.
In particular it should be published to maven central.
Some technical things:
The connector makes use of EGit internal classes. EGit has removed it from x-friends in MANIFEST.MF. Expect a lot of PDE warnings about discouraged accesses.
I would expect that these accesses to internal classes would be analyzed and be removed eventually. We'd have to figure out what stable APIs EGit core and UI would have to provide instead.
EGit does not guarantee anything about internal classes.
Bundle org.eclipse.egit.mylyn.ui, which was removed in the EGit repository in commit 12a21a6c, should probably also be moved to Mylyn.
There, too, we have internal accesses.
Cheers,
Thomas

**Comment by ruspl-afed** [comment]
Thank you for this quote @BeckerFrank , so we will need to solve it another way, the options could be:
* open PRs for EGit to fix what we need
* fork to Mylyn and do what we need
* implement our own

**Comment by merks** [comment]
There was talk about moving it:

https://www.eclipse.org/forums/index.php/m/1852831/?srch=org.eclipse.egit.github.core#msg_1852831](https://www.eclipse.org/forums/index.php?t=msg&th=1110823&goto=1852831&#msg_1852831)

But that hasn't happened.  If they do plan on maintaining it, I'm perfectly happy with that, but we really need it to to be in the main EGit update site...

I'll post a link to this issue on the EGit forum to ask again...

**Comment by BeckerFrank** [comment]
@merks: Can we close this issue?

**Comment by merks** [comment]
I can’t get much of a response. It appears to be almost dead without life support. It would be better to live somewhere where it was maintained.

**Comment by ruspl-afed** [comment]
> @merks: Can we close this issue?

I'm afraid that closing will not be enough. We need to solve it.

My vote is to grab it to Mylyn in the beginning of [4.3.0](https://github.com/eclipse-mylyn/org.eclipse.mylyn/milestone/6)

**Comment by wimjongman** [comment]
+1

**Comment by ruspl-afed** [comment]
Dear Participants,
We are near 4.3.0 M1 already
Is anyone going to work on this?

**Comment by ruspl-afed** [comment]
Too late for 4.3.0, we are preparing RC1 soon

**Comment by apupier** [comment]
Any news on this plugin?

seem the last release on maven is from 2013 https://mvnrepository.com/artifact/org.eclipse.mylyn.github/org.eclipse.egit.github.core

**Comment by ruspl-afed** [comment]
> Any news on this plugin?

This plugin is open source, anyone can revive it anywhere according to its `EPL-2.0` license.
Publishing to Maven is a different story, perhaps someone would like to take the time and set up publishing to Maven for Mylyn plugins.

**Comment by gnl42** [comment]
As discussed in #1197, I'll move the egit github code to the Mylyn repository and we can take it from there.

What the long term plan, maintain the code or switch to something like hub4j,  be can be discussed later on
